File: backend/model/Event.js

const mongoose = require('mongoose');


const  EventSchema = new mongoose.Schema({
  title: { type: String, required: true },
  location: { type: String, required: true, unique: true },
  Category: { type: mongoose.Schema.Types.ObjectId, required: true,ref:'Category' },
  image:{type:String},
  orgnaizer:{type: mongoose.Schema.Types.ObjectId, required: true,ref:'User'},
  Startdate:{type:Date,default:Date.now },
  Enddate:{type:Date,default:Date.now }



  
},{timestamps:true});



const Event = mongoose.model(' Rsvp',  EventSchema);
module.exports =  Event;

File: myapp/lib/features/Event/domain/entities/event.dart
class Event {
  final String title;
  final String location;
  final String category; 
  final String image;
  final String organizer;
  final DateTime startDate; 
  final DateTime endDate; 

  Event({
    required this.title,
    required this.location,
    required this.category,
    required this.image,
    required this.organizer,
    required this.startDate,
    required this.endDate,
  });
}
